code/game/gamemodes/umbrage/umbrage_major_abilities.dm

```diff
@@ -0,0 +1,109 @@
+	for(var/stage in 1 to 3)
+		switch(stage)
+			if(1)
+				user.visible_message("<span class='userdanger'>Vibrations pass through the air. [user]'s eyes begin to glow a deep violet.</span>", \
+									"<span class='velvet'>Psi floods into your consciousness. You feel your mind growing more powerful... <i>expanding.</i></span>")
+				playsound(user, 'sound/magic/divulge_01.ogg', 70, 0)
+			if(2)
+				user.visible_message("<span class='userdanger'>Gravity fluctuates. Psychic tendrils extend outward and feel blindly around the area.</span>", \
+									"<span class='velvet'>Gravity around you fluctuates. You tentatively reach out, feel with your mind.</span>")
+				for(var/atom/A in view(3, user))
+					spawn(rand(10, 40))
+						if(prob(25))
+							user.Beam(A, icon_state="b_beam", time = rand(30, 100))
+				playsound(user, 'sound/magic/divulge_02.ogg', 80, 0)
+			if(3)
+				user.visible_message("<span class='userdanger'>Sigils form along [user]'s body. \His skin blackens as \he glows a blinding purple.</span>", \
+									"<span class='velvet'>Your body begins to warp. Sigils etch themselves upon your flesh.</span>")
+				animate(user, color = list(rgb(0, 0, 0), rgb(0, 0, 0), rgb(0, 0, 0), rgb(0, 0, 0)), time = 200) //Produces a slow skin-blackening effect
+				playsound(user, 'sound/magic/divulge_03.ogg', 90, 0)
+		if(!do_after(user, 150, target = user))
+			user.visible_message("<span class='warning'>[user] falls to the ground!</span>", "<span class='userdanger'>Your transformation was interrupted!</span>")
+			animate(user, color = initial(user.color), pixel_y = initial(user.pixel_y), time = 10)
+			return
+	playsound(user, 'sound/magic/divulge_ending.ogg', 100, 0)
+	user.visible_message("<span class='userdanger'>[user] rises into the air, crackling with power!</span>", "<span class='velvet bold'>Your mind...! can't--- THINK--</span>")
+	animate(user, pixel_y = user.pixel_y + 5, time = 60)
+	sleep(45)
+	for(var/i in 1 to 20)
+		user << "<span class='velvet bold'>[pick("I- I- I-", "Mind-", "Sigils-", "Can't think-", "<i>POWER-</i>", "W-EE--EEE-", "<i>TAKE-</i>", "M-M-MOOORE-")]</span>"
+		sleep(1.1) //Spooky flavor message spam
+	user.visible_message("<span class='userdanger'>A tremendous shockwave emanates from [user]!</span>", "<span class='velvet big'><b>YOU ARE FREE!!</b></span>")
+	playsound(user, 'sound/magic/divulge_end.ogg', 100, 0)
+	user.fully_heal()
+	user.underwear = "Nude"
+	user.undershirt = "Nude"
+	user.socks = "Nude"
+	user.set_species(/datum/species/umbrage)
+	animate(user, color = initial(user.color), pixel_y = initial(user.pixel_y), time = 10)
+	for(var/mob/living/L in view(7, user))
+		if(L == user)
+			continue
+		L.flash_act(1, 1)
+		L.Weaken(5)
+	var/umbrage_name = pick(umbrage_names)
+	var/processed_message = "<span class='velvet'><b>\[Mindlink\] [user.real_name] has removed their human disguise and is now [umbrage_name].</b></span>"
+	for(var/datum/mind/M in ticker.mode.umbrages_and_veils)
```

Comment:
slightly faster and shows that the list you're accessing ONLY has types or subtypes of the cast in the body.

eg:
```
for(var/a in poorly_named_list)
    var/atom/A = a //for it to be safe to make this an istypeless forloop, all of the elements must be of this type
```
but:
```
for(var/m in a_list_of_mobs_and_objects)
    var/mob/M = m //Bad!! not safe!!
```